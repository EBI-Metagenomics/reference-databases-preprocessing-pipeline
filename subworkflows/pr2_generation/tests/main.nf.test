nextflow_workflow {

    name "Test Workflow PR2_GENERATION"
    script "subworkflows/pr2_generation/main.nf"
    workflow "PR2_GENERATION"
    tag "subworkflow"
    tag "pr2_generation"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                pr2_version = "5.0.0"
                pr2_label = "PR2"
                pr2_download_fasta = "https://github.com/pr2database/pr2database/releases/download/v${pr2_version}/pr2_version_${pr2_version}_SSU_mothur.fasta.gz"
                pr2_download_tax = "https://github.com/pr2database/pr2database/releases/download/v${pr2_version}/pr2_version_${pr2_version}_SSU_mothur.tax.gz"
                pr2_tax_header = "${projectDir}/subworkflows/pr2_generation/tests/data/pr2_tax_header.txt"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.fasta).match("fasta") },
                { assert snapshot(workflow.out.tax).match("tax") },
                { assert snapshot(workflow.out.otu).match("otu") }
            )
        }

    }

}
