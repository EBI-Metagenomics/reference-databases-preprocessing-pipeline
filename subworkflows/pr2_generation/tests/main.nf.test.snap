{
    "otu": {
        "content": [
            [
                "PR2.otu:md5,6e18763fa9f024468b991facdfa96e84"
            ]
        ],
        "timestamp": "2023-12-13T12:01:35.20565948"
    },
    "tax": {
        "content": [
            [
                "PR2-tax.txt:md5,2876308bd972a9e1247c9070ce3c578e"
            ]
        ],
        "timestamp": "2023-12-13T12:01:31.747721292"
    },
    "fasta": {
        "content": [
            [
                "PR2.fasta:md5,458948f0f6697ef7cefe770b0dd1b27d"
            ]
        ],
        "timestamp": "2023-12-13T12:01:28.101122999"
    }
}